src/test/java/com/amazonaws/kinesisvideo/utils/AwsV4SignerTest.java

```diff
@@ -65,19 +211,93 @@ public void signStringTest() {
     @Test
     public void getSignatureKeyTest() {
 
-        String stringToSign = "AWS4-HMAC-SHA256\n" +
+        final String stringToSign = "AWS4-HMAC-SHA256\n" +
                 "20150830T123600Z\n" +
                 "20150830/us-east-1/iam/aws4_request\n" +
                 "f536975d06c0309214f805bb90ccff089219ecd68b2577efef23edd43b7e1a59";
 
-        byte[] signatureKeyBytes = getSignatureKey("wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY",
+        final byte[] signatureKeyBytes = getSignatureKey("wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY",
                 "20150830", "us-east-1", "iam");
 
-        String expectedSignature = "c4afb1cc5771d871763a393e44b703571b55cc28424d1a5e86da6ed3c154a4b9";
+        final String expectedSignature = "c4afb1cc5771d871763a393e44b703571b55cc28424d1a5e86da6ed3c154a4b9";
         assertEquals(expectedSignature, BinaryUtils.toHex(signatureKeyBytes));
 
-        String expectedSignatureString = "5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7";
+        final String expectedSignatureString = "5d672d79c15b13162d9279b0855cfba6789a8edb4c82c400e06b5924a6f2b5d7";
         assertEquals(expectedSignatureString, BinaryUtils.toHex(hmacSha256(stringToSign, signatureKeyBytes)));
 
     }
+
+    @Test
+    public void urlEncodeTest() {
+        final String exampleArn = "arn:aws:kinesisvideo:us-west-2:123456789012:channel/demo-channel/1234567890123";
+
+        final String expected = "arn%3Aaws%3Akinesisvideo%3Aus-west-2%3A123456789012%3Achannel%2Fdemo-channel%2F1234567890123";
+
+        final String actual = urlEncode(exampleArn);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void badUrlEncodeTest() {
+        urlEncode(null);
+    }
+
+    @Test
+    public void createCredentialScopeTest() {
+        final String expected = "20150930/us-west-2/kinesisvideo/aws4_request";
```

Comment:
Use consts? AWS4_REQUEST_TYPE = "aws4_request";
 l String SERVICE = "kinesisvideo";

Reply:
We _could_ do that, but if someone happens to accidentally mess up the constants, the tests would still pass.

Since these things are part of the specification, they can't be easily changed either way.

Do you still recommend using the constants already defined in AwsV4Signer.java?

Reply:
Discussed offline. Using the constants in the tests instead of string literals.